frontend/app-development/features/dataModelling/SchemaEditorWithToolbar/SelectedSchemaEditor.test.tsx

```diff
@@ -120,7 +124,23 @@ describe('SelectedSchemaEditor', () => {
     rerender(<SelectedSchemaEditor {...updatedProps} />);
     jest.advanceTimersByTime(AUTOSAVE_DEBOUNCE_INTERVAL_MILLISECONDS);
     await waitFor(() => expect(saveDatamodel).toHaveBeenCalledTimes(1));
-    expect(saveDatamodel).toHaveBeenCalledWith(org, app, datamodelNameMock, dataMock);
+    expect(saveDatamodel).toHaveBeenCalledWith(org, app, modelPath, dataMock);
+  });
+
+  test('Does not save the datamodel when unmounting if the datamodel has been deleted', async () => {
+    const saveDatamodel = jest.fn();
+    const queryClient = createQueryClientMock();
+    const datamodelMetadataList = jest.fn().mockImplementation(() => [
+      {
+        ...jsonMetadataMock,
+        repositoryRelativeUrl: modelPath,
+      },
+    ]);
+    queryClient.setQueryData([QueryKey.JsonSchema, org, app, datamodelNameMock], dataMock);
+    queryClient.setQueryData([QueryKey.DatamodelsJson, org, app], datamodelMetadataList);
+    render({ saveDatamodel });
+    queryClient.setQueryData([QueryKey.DatamodelsJson, org, app], []);
+    expect(saveDatamodel).not.toHaveBeenCalled();
   });
```

Comment:
Denne testen er nok ikke tilstrekkelig for å sjekke at datamodellen ikke blir lagret etter sletting. Algoritmen vi vil teste er som følger:
1. Komponenten lastes inn med en valgt datamodell
2. Den valgte datamodellen slettes
3. Komponenten lastes inn på nytt med en annen valgt datamodell uten at den forrige blir lagret

For å få til dette kan vi skrive testen på samme måte som `Autosaves when changing between models that are already present in the cache`, bare at vi i tillegg oppdaterer mellomlageret før komponenten rerendres.

Her er et ekempel på hvordan det kan se ut:
```
  it('Does not save when model is deleted', async () => {
    const saveDatamodel = jest.fn();
    const queryClient = createQueryClientMock();
    queryClient.setQueryData([QueryKey.JsonSchema, org, app, model1Path], dataMock);
    queryClient.setQueryData([QueryKey.JsonSchema, org, app, model2Path], dataMock);
    const {
      renderResult: { rerender },
    } = render({ saveDatamodel }, queryClient);
    expect(saveDatamodel).not.toHaveBeenCalled();

    const updatedProps = {
      ...defaultProps,
      modelPath: model2Path,
    };
    queryClient.setQueryData([QueryKey.DatamodelsJson, org, app], [model2MetadataJson]);
    queryClient.setQueryData([QueryKey.DatamodelsXsd, org, app], [model2MetadataXsd]);
    rerender(<SelectedSchemaEditor {...updatedProps} />);
    jest.advanceTimersByTime(AUTOSAVE_DEBOUNCE_INTERVAL_MILLISECONDS);
    await expect(waitFor(() => expect(saveDatamodel).toHaveBeenCalled())).rejects.toThrow();
  });
```
Den siste linjen er litt vrien å komme frem til. Det holder ikke å bare kalle `expect(saveDatamodel).not.toHaveBeenCalled()`. Dette fordi den blir utløst av en tidtaker, som er asynkron. Vi må derfor bruke `waitFor`, men det er jo ikke noen forandring å vente på. Den beste løsningen jeg kunne finne er derfor å kjøre `waitFor` på at `saveDatamodel` _blir kalt_ og verifisere at den feiler. Det som i praksis skjer da er testen venter til `waitFor` timer ut før den konkluderer med at funksjonen aldri ble kalt. Det er en litt treg løsning, men ryddigere enn å hardkode et tidsintervall akkurat for denne testen. Jeg ville også flyttet denne ut i en egen funksjon i `testUtils` og gitt den et beskrivende navn, f.eks. `verifyNeverOccurs`. Les mer her: https://stackoverflow.com/questions/68118941/how-to-wait-for-something-not-to-happen-in-testing-library